tests/workbenches/utils.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright Skodjob authors.
+# License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+#
+from __future__ import annotations
+
+import io
+import pathlib
+from contextlib import contextmanager
+from typing import Generator
+
+from pytest_testconfig import config as py_config
+
+from kubernetes.dynamic import DynamicClient, Resource, ResourceInstance
+
+from ocp_resources.route import Route
+from ocp_resources.notebook import Notebook
+
+from utilities.constants import INTERNAL_IMAGE_REGISTRY_PATH
+
+
+def get_notebook_image(image_name: str, image_tag: str) -> str:
+    controllers_namespace = py_config["applications_namespace"]
+    if py_config.get("distribution") == "upstream":
+        image_dict = {"jupyter-minimal-notebook": "jupyter-minimal-notebook"}
+    else:
+        image_dict = {"jupyter-minimal-notebook": "s2i-minimal-notebook"}
+    return f"{INTERNAL_IMAGE_REGISTRY_PATH}/{controllers_namespace}/{image_dict[image_name]}:{image_tag}"
+
+
+def load_default_notebook(dyn_client: DynamicClient, namespace: str, name: str, image: str) -> Notebook:
+    notebook_string = (pathlib.Path(__file__).parent / "notebook-controller/test_data/notebook.yaml").read_text()
+    notebook_string = notebook_string.replace("my-project", namespace).replace("my-workbench", name)
+    # Set new Route url
+    route_name = "odh-dashboard" if py_config.get("distribution") == "upstream" else "rhods-dashboard"
+    route_host: str = list(
+        Route.get(dyn_client=dyn_client, name=route_name, namespace=py_config["applications_namespace"])
+    )[0].host
+    notebook_string = notebook_string.replace("odh_dashboard_route", "https://" + route_host)
+    # Set the correct username
+    username = get_username(dyn_client=dyn_client)
+    notebook_string = notebook_string.replace("odh_user", username)
+    # Replace image
+    notebook_string = notebook_string.replace("notebook_image_placeholder", image)
+
+    return Notebook(yaml_file=io.StringIO(notebook_string))
```

Comment:
Resource accept `kind_dict`, it's better to work with dict then with string.
You can load the YAML file with `yaml`, change the needed values inside the dict and use it to create your resource.

Reply:
At that point I can just as well follow other recommendations here to have the file content be a python literal in code...

I guess I'll do that.

Reply:
went with the kind_dict, but kept using the yaml text file for the initial data

Reply:
and kept using string replacement, because modifying the dict selectively is way more work; I need to also update values inside long strings such as container args in the pod

Reply:
changed to using dict